src/pages/PlanPage/subRoutes/UpgradePlanPage/UpgradeForm/ProPlanControls/ProPlanControls.jsx

```diff
@@ -0,0 +1,107 @@
+import { zodResolver } from '@hookform/resolvers/zod'
+import { useForm } from 'react-hook-form'
+import { useParams } from 'react-router-dom'
+
+import {
+  useAccountDetails,
+  useAvailablePlans,
+  usePlanData,
+} from 'services/account'
+import { useProPlans } from 'shared/utils/billing'
+import {
+  getDefaultValuesProUpgrade,
+  getSchema,
+  MIN_NB_SEATS_PRO,
+} from 'shared/utils/upgradeForm'
+import TextInput from 'ui/TextInput'
+
+import BillingControls from './BillingControls'
+import TotalBanner from './TotalBanner'
+import UserCount from './UserCount'
+
+import { useUpgradeControls } from '../hooks'
+import UpdateButton from '../UpdateButton'
+
+function ProPlanControls() {
+  const { provider, owner } = useParams()
+  const { data: accountDetails } = useAccountDetails({ provider, owner })
+  const { data: plans } = useAvailablePlans({ provider, owner })
+  const { data: planData } = usePlanData({ owner, provider })
+  const { proPlanYear } = useProPlans({ plans })
+  const { upgradePlan } = useUpgradeControls()
```

Comment:
Created a new hook, `useUpgradeControls` that's meant to be a hook used by any Control component. We currently have a controls hook that combines form and random data that doens't need to be there. This is a hook that exclusively returns the mutation needed when adjusting the plan + seats, separating responsibilities.